src/utils.js

```diff
@@ -59,7 +60,7 @@ async function getConfig(file: string): Promise<ZelConfig> {
  * @param {string} file
  * @return {Promise<T>}
  */
-async function sync(repo: string, branch: string, file: string): Promise<T> {
+async function sync<T>(repo: string, branch: string, file: string): Promise<T> {
```

Comment:
It's also faster than native Promise, but not big deal here at all.

And yeah, soon I was looking to replace caporal as well. Dunno if you saw my recent `mri`, but is a good start imo.